### ISSUE TITLE
Geometry entity with translucent image material bug

### ISSUE THREAD

**Issue by hpinkos** [AnalyticalGraphicsInc/cesium#3459]
Reported on the forum: https://groups.google.com/forum/?hl=en#!topic/cesium-dev/GMCZ7y4vLbo
This was introduced in 1.16.  The translucent parts of the png are filled in black.
![image](https://cloud.githubusercontent.com/assets/3451886/12556451/ef39e840-c353-11e5-8d18-62c1fb47deec.png)

Code example: 
``` javascript
var viewer = new Cesium.Viewer('cesiumContainer');

viewer.entities.add({
    position : Cesium.Cartesian3.fromDegrees(-110, 40),
    show: true,
    ellipse : {
        semiMinorAxis : 500000,
        semiMajorAxis : 1000000,
        material : new Cesium.ImageMaterialProperty({
            image : '../images/Cesium_Logo_overlay.png'
        })
    }
});
```

I think this is happening because the `translucent` property of the appearance is being set to `false`.  Adding `alpha: 0.99` to the material makes it render correctly.

**Comment by mramato** [comment]
Yes, this is a change in behavior after #3206.  I'm not sure of the best solution, but we may need to add another property to the Entity API.  `alpha` affects the transparency of the opaque parts, while a `transparent` property would indicate if the image itself has transparency.  This is definitely sub-optimal because even the developer may not know the answer but I don't think we can detect this programatically either.

I'm open to ideas.

**Comment by mramato** [comment]
Talked with @bagnell offline, this is impossible to fix in all cases without renderer changes.  The problem is that in order for transparency to work at all, it needs to be in the transparent pass; however, putting it in the transparent pass means that volumes, such as a box or sphere, will end up with the backside rendered as well.  For example, here is master:

![image](https://cloud.githubusercontent.com/assets/1495196/12557420/a06573e2-c358-11e5-8e73-a8ae177027cf.png)

after fixing the issue by always reporting as translucent, this happens:

![image](https://cloud.githubusercontent.com/assets/1495196/12557341/491af1e8-c358-11e5-931d-29b7834dbbc0.png)

The parts of the Cesium logo that should be opaque don't appear so because the back and front sides get blended.

I don't think this changes my original assertion that we'll need a new property (and then change the default to always be translucent), but no solution right now is perfect.

**Comment by hpinkos** [comment]
@mramato  I still think the second case, reporting it as translucent, is more expected behavior than filling the transparent parts in black. Could we change it back to that for now even though it's not technically "correct"?
Alternatively, could we fix it for ellipsoid surface geometry types like `Rectangle`, `Ellipse` and `Polygon`?

Or should I just keep pointing people to use an alpha value as a workaround?

**Comment by mramato** [comment]
The problem is that changing it back means that it will be wrong for all volumes, even when images are completely opaque.  We need to both change it back and expose the property I think.

**Comment by mramato** [comment]
Setting the alpha is probably a good work around for the time being, since it's very similar to what the "fix" will do anyway.

**Comment by hpinkos** [comment]
Also mentioned here in a later post: https://groups.google.com/forum/?hl=en#!topic/cesium-dev/W7cUl_uWAZU

**Comment by denverpierce** [comment]
This also affects KML <GroundOverlay>'s with alpha channels.

**Comment by hpinkos** [comment]
Mentioned again here: https://groups.google.com/forum/?hl=en#!topic/cesium-dev/-Zdw6sO00WM

I still think we should change this so the images are transparent.  I think it's better that it's "wrong for all volumes" and be correct for ground overlays, since I think that's what most people will be using transparent pngs for.

**Comment by mramato** [comment]
@hpinkos I suggested a solution in my original reply.  We should probably add a boolean `transparent` flag so that we know whether the image itself contains transparency or not.  I'm happy to look at a pull request.

**Comment by hpinkos** [comment]
@mramato how would we handle that for DataSources?  That person from the forum was using KML

**Comment by mramato** [comment]
We would set the default back to the old behavior.

**Comment by hpinkos** [comment]
This issue was fixed in #3766 

For the Entity API, set `transparent: true` for the `ImageMaterialProperty`
```
var viewer = new Cesium.Viewer('cesiumContainer');
viewer.entities.add({
    position : Cesium.Cartesian3.fromDegrees(-110, 40),
    show: true,
    ellipse : {
        semiMinorAxis : 500000,
        semiMajorAxis : 1000000,
        material : new Cesium.ImageMaterialProperty({
            image : '../images/Cesium_Logo_overlay.png',
            transparent: true
        })
    }
});
```
